vm/rules.go

```diff
@@ -0,0 +1,179 @@
+
+package vm
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/fees"
+
+	hconsts "github.com/ava-labs/hypersdk/consts"
+)
+
+var (
+	_ chain.Rules              = (*BaseRules)(nil)
+	_ RuleFactory[chain.Rules] = (*BaseRules)(nil)
+)
+
+type BaseRules struct {
+	NetworkID uint32 `json:"networkID"`
+	ChainID   ids.ID `json:"chainID"`
+
+	// Chain Parameters
+	MinBlockGap      int64 `json:"minBlockGap"`      // ms
+	MinEmptyBlockGap int64 `json:"minEmptyBlockGap"` // ms
+
+	// Chain Fee Parameters
+	MinUnitPrice               fees.Dimensions `json:"minUnitPrice"`
+	UnitPriceChangeDenominator fees.Dimensions `json:"unitPriceChangeDenominator"`
+	WindowTargetUnits          fees.Dimensions `json:"windowTargetUnits"` // 10s
+	MaxBlockUnits              fees.Dimensions `json:"maxBlockUnits"`     // must be possible to reach before block too large
+
+	// Tx Parameters
+	ValidityWindow      int64 `json:"validityWindow"` // ms
+	MaxActionsPerTx     uint8 `json:"maxActionsPerTx"`
+	MaxOutputsPerAction uint8 `json:"maxOutputsPerAction"`
+
+	// Tx Fee Parameters
+	BaseComputeUnits          uint64   `json:"baseUnits"`
+	StorageKeyReadUnits       uint64   `json:"storageKeyReadUnits"`
+	StorageValueReadUnits     uint64   `json:"storageValueReadUnits"` // per chunk
+	StorageKeyAllocateUnits   uint64   `json:"storageKeyAllocateUnits"`
+	StorageValueAllocateUnits uint64   `json:"storageValueAllocateUnits"` // per chunk
+	StorageKeyWriteUnits      uint64   `json:"storageKeyWriteUnits"`
+	StorageValueWriteUnits    uint64   `json:"storageValueWriteUnits"` // per chunk
+	SponsorStateKeysMaxChunks []uint16 `json:"sponsorStateKeysMaxChunks"`
+}
+
+// GetRules would normally depend on the param, but these rules are unchanging
+func (r *BaseRules) GetRules(_ int64) chain.Rules {
+	return r
+}
+
+func DefaultRules() *BaseRules {
+	return &BaseRules{
+		// Chain Parameters
+		MinBlockGap:      100,
+		MinEmptyBlockGap: 750,
+
+		// Chain Fee Parameters
+		MinUnitPrice:               fees.Dimensions{100, 100, 100, 100, 100},
+		UnitPriceChangeDenominator: fees.Dimensions{48, 48, 48, 48, 48},
+		WindowTargetUnits:          fees.Dimensions{20_000_000, 1_000, 1_000, 1_000, 1_000},
+		MaxBlockUnits:              fees.Dimensions{1_800_000, 2_000, 2_000, 2_000, 2_000},
+
+		// Tx Parameters
+		ValidityWindow:      60 * hconsts.MillisecondsPerSecond, // ms
+		MaxActionsPerTx:     16,
+		MaxOutputsPerAction: 1,
+
+		// Tx Fee Compute Parameters
+		BaseComputeUnits: 1,
+
+		// Tx Fee Storage Parameters
+		//
+		// TODO: tune this
+		StorageKeyReadUnits:       5,
+		StorageValueReadUnits:     2,
+		StorageKeyAllocateUnits:   20,
+		StorageValueAllocateUnits: 5,
+		StorageKeyWriteUnits:      10,
+		StorageValueWriteUnits:    3,
+		SponsorStateKeysMaxChunks: []uint16{1},
+	}
+}
+
+func LoadBaseRules(b []byte, _ []byte /* upgradeBytes */) (*BaseRules, error) {
```

Comment:
Can we add `networkID` and `chainID` to the params here?

By default, these should be set to those values and only in extreme circumstances when a running network needs to hard fork and change their networkID / chainID would these ever change